src/components/structures/ElementProAction.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as React from "react";
+import ElementProDialog from "../views/dialogs/ElementProDialog";
+import Modal, {IHandle} from "../../Modal";
+import {
+    IconizedContextMenuOption,
+    IconizedContextMenuOptionList,
+} from "../views/context_menus/IconizedContextMenu";
+import { _t } from "../../languageHandler";
+
+interface IProps {}
+
+interface IState {}
+
+export default class ElementProAction extends React.PureComponent<IProps, IState> {
```

Comment:
I'm not sure I'm convinced that we need to call this thing "Element Pro" in code. By having it be a config, we're implying that it is a generic service that someone can run if they want to - I see no reason why we need to call that service Element Pro when we want to use that branding for something else.

Reply:
the other concern is that all of this branding is done under the Foundation, which is expected to be neutral to this sort of thing.

Reply:
Can we agree on a good component name, to avoid unnecessary commits - how about `HostSignup` for the components and `host_signup` for the config?

Reply:
The components could also be placed under some namespaced folder, if that helps.

Reply:
`HostSignup` or some variation of that seems reasonable. I was actually going to suggest `HostUpgrade` could also work.